sys/net/application_layer/unicoap/state.c

```diff
@@ -192,6 +180,19 @@ kernel_pid_t unicoap_init(void)
     UNICOAP_DEBUG("registered %" PRIuSIZE " XFA resources\n", _xfa_listener.resource_count);
 #endif
 
+    event_queue_init(&_queue);
+
+    /* The waiter thread will be in _unicoap_event_loop_run.
+     * The queue needs to be initialized to initialize drivers.
+     * As preventative measure, we set waiter to NULL until the real, but currently unknown
+     * thread is known. */
+    _queue.waiter = NULL;
```

Comment:
```suggestion
    event_queue_init_detached(&_queue);
```

Shouldn't that do the trick as mentioned before?